src/commonMain/kotlin/io/github/kevincianfarini/alchemist/unit/AreaUnit.kt

```diff
@@ -20,12 +20,31 @@ public interface AreaUnit {
     /**
      * A non-standard representation of area commonly used as part of the metric system.
      */
-    public enum class Metric(override val symbol: String, override val millimetersSquaredScale: Long) : AreaUnit {
-        Decimilliare("dma", 100),
-        Centiare("ca", 1_000_000),
-        Deciare("da", 10_000_000),
-        Are("a", 100_000_000),
-        Decare("daa", 1_000_000_000),
-        Hectare("ha", 10_000_000_000),
+    public object Metric {
+        public val Decimilliare: AreaUnit = object : AreaUnit {
+            override val symbol: String = "dma"
+            override val millimetersSquaredScale: Long = 100
+        }
+        public val Centiare: AreaUnit = object : AreaUnit {
+            override val symbol: String = "ca"
+            override val millimetersSquaredScale: Long = 1_000_000
+        }
+        public val Deciare: AreaUnit = object : AreaUnit {
+            override val symbol: String = "da"
+            override val millimetersSquaredScale: Long = 10_000_000
+        }
+        public val Are: AreaUnit = object : AreaUnit {
+            override val symbol: String = "a"
+            override val millimetersSquaredScale: Long = 100_000_000
+        }
+        public val Decare: AreaUnit = object : AreaUnit {
+            override val symbol: String = "daa"
+            override val millimetersSquaredScale: Long = 1_000_000_000
+        }
+        public val Hectare: AreaUnit = object : AreaUnit {
+            override val symbol: String = "ha"
+            override val millimetersSquaredScale: Long = 10_000_000_000
+        }
+        public val entries: List<AreaUnit> = listOf(Decimilliare, Centiare, Deciare, Are, Decare, Hectare)
```

Comment:
I think it might make sense to make all of these getters? Eg. 

```kt
private val Decimilliare: AreaUnit get() = object : AreaUnit { ... }
```

Reply:
Sure. I generally don't like that pattern because the objects lose their name and and become anonymous objects.

Reply:
Actually, on second thought -- I think we should pause this until #42 is complete. Each of these being anonymous objects is pretty awkward and this will likely significantly change once the work in the above issue is completed.